api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -393,6 +393,10 @@
      */
     Attribute<String> CONTAINER_JVM_OPTIONS = new Attribute<>(String2String.getInstance());
     /**
+     * The options of dynamic apex logger appender
+     */
+    Attribute<String> LOGGER_APPENDER = new Attribute<>(String2String.getInstance());
```

Comment:
The log messages on the launch side are also important and need to be collected to get the end-to-end picture. This is because apart from the engine logs, user code also runs on the launch side and may have log messages. In that case, this setting should also be available as a global setting, outside of a per-application basis. Take a look at SPIN_MILLIS which is both a port setting and an operator setting. Also, consider the scenario where a user specifies dt.application.app1.attr.APPENDER and dt.application.app2.attr.APPENDER which appender would be used for logs before the application is selected and launched?

The second thing is the logger levels are specified as a global setting StramUtils.DT_LOGGERS_LEVEL and is different from this specification. Consider bringing them to a consistent specification.